libcnb/src/layer/test.rs

```diff
@@ -0,0 +1,898 @@
+
+    // Assert basic LayerData fields
+    assert_eq!(
+        &layer_content_metadata_from_disk,
+        &expected_content_layer_metadata
+    );
+
+    assert_eq!(handle_layer_result.name, layer_name);
+    assert_eq!(
+        handle_layer_result.path,
+        temp_dir.path().join("layers").join(layer_name.as_str())
+    );
+
+    assert_eq!(handle_layer_result.env, LayerEnv::new());
+
+    // Assert layer directory contents
+    let create_file_contents =
+        fs::read_to_string(handle_layer_result.path.join(TEST_LAYER_CREATE_FILE_NAME)).ok();
+
+    let update_file_contents =
+        fs::read_to_string(handle_layer_result.path.join(TEST_LAYER_UPDATE_FILE_NAME)).ok();
+
+    assert_eq!(
+        create_file_contents,
+        Some(String::from(TEST_LAYER_CREATE_FILE_CONTENTS))
+    );
+
+    assert_eq!(update_file_contents, None);
+}
+
+#[test]
+fn error_handling_no_metadata_toml() {
+    let temp_dir = tempdir().unwrap();
+    let context = build_context(&temp_dir);
+    let layer_name = random_layer_name();
+
+    let layer_dir = temp_dir.path().join("layers").join(layer_name.as_str());
+    fs::create_dir_all(&layer_dir).unwrap();
+
+    let handle_layer_result = handle_layer(&context, layer_name, TestLayer::default()).unwrap();
+
+    let create_file_contents =
+        fs::read_to_string(handle_layer_result.path.join(TEST_LAYER_CREATE_FILE_NAME)).ok();
+
+    let update_file_contents =
+        fs::read_to_string(handle_layer_result.path.join(TEST_LAYER_UPDATE_FILE_NAME)).ok();
+
+    assert_eq!(
+        create_file_contents,
+        Some(String::from(TEST_LAYER_CREATE_FILE_CONTENTS))
+    );
+
+    assert_eq!(update_file_contents, None);
+}
+
+#[test]
+fn error_handling_no_directory() {
+    let temp_dir = tempdir().unwrap();
+    let context = build_context(&temp_dir);
+    let layer_name = random_layer_name();
+
+    let layer_toml_path = temp_dir
+        .path()
+        .join("layers")
+        .join(format!("{}.toml", layer_name.as_str()));
+
+    fs::write(
+        &layer_toml_path,
+        r#"
+[types]
+launch = true
+build = true
+cache = true
+
+[metadata]
+version = "3.2.1"
+    "#,
+    )
+    .unwrap();
+
+    let handle_layer_result = handle_layer(&context, layer_name, TestLayer::default()).unwrap();
+
+    // We expect the layer to be recreated from scratch. This means that the version from the
+    // existing metadata should not be used.
+    assert_ne!(
+        handle_layer_result.content_metadata.metadata.version,
+        "3.2.1"
+    );
+
+    let create_file_contents =
+        fs::read_to_string(handle_layer_result.path.join(TEST_LAYER_CREATE_FILE_NAME)).ok();
+
+    let update_file_contents =
+        fs::read_to_string(handle_layer_result.path.join(TEST_LAYER_UPDATE_FILE_NAME)).ok();
+
+    assert_eq!(
+        create_file_contents,
+        Some(String::from(TEST_LAYER_CREATE_FILE_CONTENTS))
+    );
+
+    assert_eq!(update_file_contents, None);
+}
+
+#[test]
+fn write_layer_env() {
```

Comment:
This test tests that the env is written to disk, but do we also need a test confirming that a previously written env is read back in successfully on next `handle_layer()` (ie check the `LayerEnv::read_from_layer_dir()` codepath inside `read_layer()`)? 

As far as I can see none of the other tests cover this case so far.

Reply:
Very good point!

Reply:
Added a test in aaf730d